src/player/Player.php

```diff
@@ -1544,6 +1544,15 @@ public function onUpdate(int $currentTick) : bool{
 				$this->fireTicks = 1;
 			}
 
+			$item = $this->getInventory()->getItemInHand();
+			if($item instanceof Releasable && $this->isUsingItem()){
+				$oldItem = clone $item;
+				if($item->continueUsing($this, $this->getItemUseDuration())){
+					$this->getNetworkSession()->onChargeItemComplete();
+					$this->returnItemsFromAction($oldItem, $item, []);
```

Comment:
We should probably do this irrespective of the result of continueUsing(). For example, a custom item might have the crossbow string snap and break the item during usage, which wouldn't count as charging it, but would still require the inventory to be updated.